dbt/include/athena/macros/materializations/models/table/table.sql

```diff
@@ -133,20 +133,27 @@
         {%- endif -%}
 
         {%- set old_relation_table_type = adapter.get_glue_table_type(old_relation) -%}
+        -- we cannot use old_bkp_relation, because it returns None if the relation doesn't exist
+        -- we need to create a python object via the make_temp_relation instead
+        {%- set old_relation_bkp = make_temp_relation(old_relation, '__bkp') -%}
 
-        {%- if old_relation_table_type == 'iceberg' -%}
-          {{ rename_relation(old_relation, old_bkp_relation) }}
+        {%- if old_relation_table_type.value == 'iceberg_table' -%}
+          {{ rename_relation(old_relation, old_relation_bkp) }}
         {%- else  -%}
           {%- do drop_relation_glue(old_relation) -%}
         {%- endif -%}
 
         {{ rename_relation(tmp_relation, target_relation) }}
 
-        -- old_bkp_relation might not exists in case we have a switch from hive to iceberg
-        -- we prevent to drop something that doesn't exist even if drop_relation is able to deal with not existing tables
-        {%- if old_bkp_relation is not none -%}
-          {%- do drop_relation(old_bkp_relation) -%}
+        -- old_relation_bkp might not exists in case we have a switch from hive to iceberg
```

Comment:
If we are sure that the target relation has been updated at this point why not simply issue a `DROP IF EXISTS` and avoid this call altogether?

Reply:
@Avinash-1394, that would occur if the `native_drop`is set, but wouldnt work for direct glue calls

Reply:
@Avinash-1394, that would occur if the native_dropis set, but wouldnt work for direct glue calls

Reply:
Just to make clear why we have 2 behaviours for dropping, when you run `DROP IF EXISTS` for iceberg tables, the location is left dirty in case of big tables, therefore our drop relation implementation uses glue apis with s3 apis to drop a table.

Also there is an edge case where we switch from hive to iceberg, where old_bkp_relation doesn't not exist.